File: zzsong-iam-server/src/main/kotlin/com/zzsong/iam/server/infrastructure/encoder/password/PasswordEncoder.java
package com.zzsong.iam.server.infrastructure.encoder.password;

import javax.annotation.Nonnull;

/**
 * @author 宋志宗 on 2021/7/5
 */
public interface PasswordEncoder {

  String encode(@Nonnull CharSequence rawPassword);

  boolean matches(@Nonnull CharSequence rawPassword, @Nonnull String encodedPassword);
}

File: zzsong-iam-user/src/main/kotlin/com/zzsong/iam/user/infrastructure/password/encoder/PasswordEncoder.kt
package com.zzsong.iam.user.infrastructure.password.encoder

import javax.annotation.Nonnull

/**
 * @author 宋志宗 on 2021/7/5
 */
interface PasswordEncoder {

  fun encode(rawPassword: CharSequence): String

  fun matches(rawPassword: CharSequence, @Nonnull encodedPassword: String): Boolean
}
